kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java

```diff
@@ -285,42 +248,73 @@ private void getAllChannelsByAssetPid(final KuraResponsePayload respPayload, fin
+    }
+
     /**
-     * Prepares the response payload based on the asset records as provided
-     *
+     * Put operation specific for assets
+     * 
+     * @param reqPayload
+     *            a KuraRequestPayload
      * @param respPayload
-     *            the response payload to prepare
-     * @param assetRecords
-     *            the list of asset records
-     * @throws NullPointerException
-     *             if any of the arguments is null
+     *            a KuraResponsePayload
+     * @param resources
+     *            String array representing the topic chunks
      */
-    private void prepareResponse(final KuraResponsePayload respPayload, final List<AssetRecord> assetRecords) {
-        requireNonNull(respPayload, s_message.respPayloadNonNull());
-        requireNonNull(assetRecords, s_message.assetRecordsNonNull());
-
-        for (final AssetRecord assetRecord : assetRecords) {
-            final TypedValue<?> assetValue = assetRecord.getValue();
-            final String value = (assetValue != null) ? String.valueOf(assetValue.getValue()) : "ERROR";
-            String errorText;
-            final AssetStatus assetStatus = assetRecord.getAssetStatus();
-            final AssetFlag assetFlag = assetStatus.getAssetFlag();
-
-            final String prefix = assetRecord.getChannelId() + ".";
-            respPayload.addMetric(prefix + s_message.flag(), assetFlag.toString());
-            respPayload.addMetric(prefix + s_message.channel(), assetRecord.getChannelId());
-            respPayload.addMetric(prefix + s_message.timestamp(), assetRecord.getTimestamp());
-            respPayload.addMetric(prefix + s_message.value(), value);
+    private void doPutAssets(final KuraRequestPayload reqPayload, final KuraResponsePayload respPayload,
+            final String[] resources) {
+        this.findAssets();
+        final String assetPid = resources[1];
+        final String channelId = resources[2];
+        final Asset asset = this.assets.get(assetPid);
+        final AssetConfiguration configuration = asset.getAssetConfiguration();
+        final Map<Long, Channel> assetConfiguredChannels = configuration.getAssetChannels();
+        final long id = Long.parseLong(channelId);
+        if ((assetConfiguredChannels != null) && (id != 0)) {
+            final AssetRecord assetRecord = new AssetRecord(id);
+            final String userValue = (String) reqPayload.getMetric("value");
+            final String userType = (String) reqPayload.getMetric("type");
+            boolean flag = true;
+            try {
+                this.wrapValue(assetRecord, userValue, userType);
+            } catch (final NumberFormatException nfe) {
+                flag = false;
+                assetRecord.setAssetStatus(new AssetStatus(AssetFlag.FAILURE, message.valueTypeConversionError(), nfe));
+                assetRecord.setTimestamp(System.currentTimeMillis());
+            }
 
-            if (assetFlag == AssetFlag.FAILURE) {
```

Comment:
According to this, if any error happens, there won't be any response sent to the user. The user will keep waiting for it with hope of something to appear 😉